zebra-test/src/command.rs

```diff
@@ -248,20 +185,130 @@ impl TestOutput {
             return Ok(self);
         }
 
-        let command = || self.cmd.clone().header("Command:");
-        let stdout = || stdout.into_owned().header("Stdout:");
+        Err(eyre!("stdout of command is not equal to the given regex")).context_from(self)
+    }
+
+    /// Returns Ok if the program was killed, Err(Report) if exit was by another
+    /// reason.
+    pub fn assert_was_killed(&self) -> Result<()> {
+        if self.was_killed() {
+            Err(eyre!("command was killed")).context_from(self)?
+        }
+
+        Ok(())
+    }
+
+    /// Returns Ok if the program was not killed, Err(Report) if exit was by
+    /// another reason.
+    pub fn assert_was_not_killed(&self) -> Result<()> {
+        if !self.was_killed() {
+            Err(eyre!("command wasn't killed")).context_from(self)?
+        }
 
-        Err(eyre!("stdout of command is not equal to the given regex"))
-            .with_section(command)
-            .with_section(stdout)
+        Ok(())
     }
 
-    /// Returns true if the program was killed, false if exit was by another reason.
-    pub fn was_killed(&self) -> bool {
-        #[cfg(unix)]
-        return self.output.status.signal() == Some(9);
+    #[cfg(not(unix))]
+    fn was_killed(&self) -> bool {
+        self.output.status.code() != Some(1)
+    }
 
-        #[cfg(not(unix))]
-        return self.output.status.code() == Some(1);
+    #[cfg(unix)]
+    fn was_killed(&self) -> bool {
+        self.output.status.signal() != Some(9)
+    }
+}
+
+/// Add context to an error report
+trait ContextFrom<S> {
+    type Return;
+
+    fn context_from(self, source: &S) -> Self::Return;
+}
+
+impl<C, T, E> ContextFrom<C> for Result<T, E>
+where
+    E: Into<Report>,
+    Report: ContextFrom<C, Return = Report>,
+{
+    type Return = Result<T, Report>;
+
+    fn context_from(self, source: &C) -> Self::Return {
+        self.map_err(|e| e.into())
+            .map_err(|report| report.context_from(source))
+    }
+}
```

Comment:
This feels intensely big brain to me and legit solved a problem esteban and I had tried to solve 6 months ago in `color-eyre` and had collectively failed to figure out, aka how to have an extension trait that is implemented for `a type` and `Result<T, that type>`. I just casually solved it this time somehow and ended up going back to color-eyre and applying this same fix there so now you can call `with_section` on a `Result` or directly on a `Report`

:pray: so happy rn

Reply:
This is very nice 🤩